third-party/fedora33/qemu/BUCK

```diff
@@ -1,45 +1,126 @@
+
+# these need to be updated manually based on the downloaded version
+qemu = {
+    'version': '6.0.0',
+    'sha256': '87bc1a471ca24b97e7005711066007d443423d19aacda3d442558ae032fa30b9',
+}
+pixman = {
+    'version': '0.40.0',
+    'sha256': '3a68a28318a78fffc61603c8385bb0010c3fb23d17cd1285d36a7148c87a3b91',
+}
+libcap_ng = {
+    'version': '0.8.2',
+    'sha256': '65b86885b8d873e55c05bd49427fd370d559b26f0c2089ac9194828e6a2fe233',
+}
+
+export_file(name = 'build.sh')
+export_file(name = 'meson.build.patch')
+export_file(name = 'libcap-ng.pc.patch')
+
+http_file(
+    name = 'source',
+    urls = ['https://download.qemu.org/qemu-{}.tar.xz'.format(qemu['version'])],
+    sha256 = qemu['sha256'],
+)
+
+# packages which don't provide static libs also need to be built from source
+http_file(
+    name = 'pixman',
+    urls = ['https://github.com/freedesktop/pixman/archive/refs/tags/pixman-{}.tar.gz'.format(pixman['version'])],
+    sha256 = pixman['sha256'],
+)
+http_file(
+    name = 'libcap-ng',
+    urls = ['https://github.com/stevegrubb/libcap-ng/archive/refs/tags/v{}.tar.gz'.format(libcap_ng['version'])],
+    sha256 = libcap_ng['sha256'],
+)
+
+image.layer(
+    name = '_base',
+    parent_layer = '//images/appliance:stable-build-appliance',
+    features = [
+        image.rpms_install([
+            # dependencies listed at https://wiki.qemu.org/Hosts/Linux
+            'git', 'bzip2', 'python3', 'ninja-build',
+            'glib2-devel', 'libfdt-devel', 'pixman-devel', 'zlib-devel',
+            # static build dependencies (at least on this builder image)
+            'glib2-static', 'glibc-static', 'pcre-static', 'zlib-static',
+            'libzstd-static', 'autoconf', 'libtool',
+            # needed for virtfs
+            'libcap-ng-devel', 'libattr-devel',
+        ]),
+    ],
+)
+
+image.layer(
+    name = '_setup',
+    parent_layer = ':_base',
+    features = [
+        # genrule_layer runs as user 'nobody' by default
+        image.ensure_subdirs_exist('/', '_temp_qemu', user='nobody'),
+        image.ensure_subdirs_exist('/', 'output', user='nobody'),
+
+        # put sources and build script in the base layer
+        image.install(':build.sh', '/_temp_qemu/build.sh'),
+        image.install(':source', '/_temp_qemu/source.tar.xz'),
+        image.install(':pixman', '/_temp_qemu/pixman.tar.gz'),
+        image.install(':libcap-ng', '/_temp_qemu/libcap-ng.tar.gz'),
+        image.install(':meson.build.patch', '/_temp_qemu/meson.build.patch'),
+        image.install(':libcap-ng.pc.patch', '/_temp_qemu/libcap-ng.pc.patch'),
     ],
 )
 
-export_file(
-    name = "qemu-img",
-    visibility = [
-        "//antlir/linux/...",
-        "//antlir/vm/...",
-        "//third-party/...",
+image.genrule_layer(
+    name = '_build',
+    parent_layer = ':_setup',
+    rule_type = 'build',
+    antlir_rule = 'user-internal',
+    # user = 'root',  # needed to patch .pc files, sometimes for untarring
+    cmd = [
+        '/bin/sh', '-c',
```

Comment:
I don't think you really need a shell for this, you can just `pushd` into `/_temp_qemu` in the beginning of `build.sh` and give the version arguments as array elements after `/_temp_qemu/build.sh`